lib/matplotlib/backends/registry.py

```diff
@@ -0,0 +1,60 @@
+from enum import Enum
+
+
+class BackendFilter(Enum):
+    INTERACTIVE = 0
+    INTERACTIVE_NON_WEB = 1
```

Comment:
Why does ipython need to know about what backends are "web" or not?  Is that something that could be encoded in the required_interactive_framework attribute (e.g. by declaring a separate "web" value)?
This is mostly out of concern as to whether third-party backends ought to be able to declare themselves as "web" backends or not.

Reply:
This is nothing to do with IPython. This is a rearrangement of the existing code without any functional change. The 2 web-based backends were hardcoded in `pyplot.py` somewhere around line 2500, and that hard-coding has moved to this file.